core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -0,0 +1,137 @@
+          //        from the request.
+          if (NoSuchNamespaceException.class.getSimpleName().equals(error.type())) {
+            throw new NoSuchNamespaceException("Resource not found: %s", error);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", error);
+          }
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("The table already exists: %s", error);
+        default:
+          throw new RESTException("Unknown error: %s", error);
+      }
+    };
+  }
+
+  /**
+   * Request error handlers specifically for CRUD ops on namespaces.
+   * Should be chained wih the {@linkplain ErrorHandlers#defaultErrorHandler}, which takes care of common cases.
+   */
+  public static Consumer<ErrorResponse> namespaceErrorHandler() {
+    return error -> {
+      switch (error.code()) {
+        case HttpStatus.SC_NOT_FOUND:
+          throw new NoSuchNamespaceException("Namespace not found: %s", error);
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("Namespace already exists: %s", error);
+        // Invalid UpdateNamespacePropertiesRequest - keys to remove overlapped with keys of properties to add
+        case HttpStatus.SC_UNPROCESSABLE_ENTITY:
+          throw new RESTException("Unable to process request due to bad property updates");
+        case HttpStatus.SC_FORBIDDEN:
+        case HttpStatus.SC_UNAUTHORIZED:
+          throw new NotAuthorizedException("Not Authorized: %s", error);
+        default:
+          throw new RESTException("Unknown error: %s", error);
+      }
+    };
+  }
+
+  /**
+   * Request error handler that handles the common cases that are included with all responses,
+   * such as 400, 401, 403, 500, etc.
+   */
+  public static Consumer<ErrorResponse> defaultErrorHandler() {
+    return errorResponse -> {
+      switch (errorResponse.code()) {
+        // 400
+        case HttpStatus.SC_CLIENT_ERROR:
+          throw new BadRequestException("Unable to process the request as it is somehow malformed: %s", errorResponse);
+        // 401
+        case HttpStatus.SC_UNAUTHORIZED:
+          throw new NotAuthorizedException("Not Authorized: %s", errorResponse);
+        // 403
+        case HttpStatus.SC_FORBIDDEN:
+          throw new ForbiddenException("Forbidden: %s", errorResponse);
+        // 405 & 406 & 501 - All of these can represent not supported.
+        case HttpStatus.SC_METHOD_NOT_ALLOWED:
+        case HttpStatus.SC_NOT_ACCEPTABLE:
+        case HttpStatus.SC_NOT_IMPLEMENTED:
+          throw new UnsupportedOperationException(String.format("Not supported: %s", errorResponse));
+        // 500
+        case HttpStatus.SC_SERVER_ERROR:
+          throw new ServiceFailureException("Server error: %s", errorResponse);
+        default:
+          throw new RESTException("Unknown error: %s", errorResponse);
+      }
+    };
+  }
+
+  static String extractResponseBodyAsString(CloseableHttpResponse response) {
```

Comment:
Is this used? I thought this part of the API was going to avoid depending on a specific HTTP library.

Reply:
Looks like this should be moved into `HttpRESTClient`.